### PR TITLE
Feature/table

### PR DESCRIPTION
**Pull request by akinolu52**
## What
*Creates a Table component to handle pagination, sorting and filtering of table data*

## How
*I implemented a custom table component that handles the various operations needed on a table data*

### CODE REVIEW

File: re_data_ui/public/index.html [hunk 1]
```diff
@@ -1,43 +1,24 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <meta name="theme-color" content="#000000" />
+  <meta name="description"
+    content="re_data is a web application that allows you to visualize your data in a more intuitive way." />
```
Comment (mateuszklimek) [inline re_data_ui/public/index.html:11]: Let's put line from github docs there: `re_data is an open-source data quality & observability framework for the modern data stack.`

Comment (akinolu52) [reply re_data_ui/public/index.html:11]: okau

Comment (akinolu52) [reply re_data_ui/public/index.html]: updated

https://github.com/re-data/re-data/pull/173/commits/650c503a6f8eaa9e629095a05d4df2735b4c775c

File: re_data_ui/tsconfig.json [hunk 2]
```diff
@@ -19,14 +19,18 @@
...
+    "typeRoots": [
+      "./types",
+      "./node_modules/@types"
+    ],
     "downlevelIteration": true,
   },
   "include": [
-    "src"
+    "src",
+    "types/react-table/index.d.ts"
```
Comment (dejii) [inline re_data_ui/tsconfig.json:30]: there are some issues with ts detecting the types from the `@types/react-table` package?

File: re_data_ui/src/pages/Alerts.tsx [hunk 3]
```diff
@@ -1,124 +1,122 @@
...
+  ],
+  []);
+  const data = useMemo(() => generateAlertData(alerts), [alerts]) || [];
+
   return (
     <>
       {(alerts.length || !graph)
         ? (
-          <div className="grid grid-cols-1">
+          <div className="grid grid-cols-1" style={{ overflowY: 'scroll' }}>
```
[the remark this answers is not in the document]
Comment (akinolu52) [reply re_data_ui/src/pages/Alerts.tsx:116]: okay, thanks

File: re_data_ui/src/pages/Alerts.tsx [hunk 4]
```diff
@@ -1,124 +1,122 @@
-import React, { ReactElement, useContext } from 'react';
-import { Link } from 'react-router-dom';
+/* eslint-disable comma-dangle */
```
[the remark this answers is not in the document]
Comment (akinolu52) [reply re_data_ui/src/pages/Alerts.tsx:1]: it's my code editor that removes that last comma,  I've removed the eslint comment

[the remark this answers is not in the document]
Comment (dejii) [reply re_data_ui/src/pages/Alerts.tsx:1]: 👍

[the remark this answers is not in the document]
Comment (dejii) [reply re_data_ui/src/pages/Alerts.tsx:1]: you can configure your project to adhere to the eslint config as well so it doesn't remove some things.

File: re_data_ui/tsconfig.json (hunk 2 above)
Comment (akinolu52) [reply re_data_ui/tsconfig.json:30]: removed(it was added by my editor)

Comment (dejii) [reply re_data_ui/tsconfig.json:30]: okay, don't forget to update the pr.

Comment (akinolu52) [reply re_data_ui/tsconfig.json]: done

File: re_data_ui/src/pages/Alerts.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (akinolu52) [reply re_data_ui/src/pages/Alerts.tsx]: okay, i'll check that out.

**Review by dejii — COMMENTED** [review]
looks good to me 👍

**Pull request merged**